core/src/test/java/hudson/util/io/ZipArchiverTest.java

```diff
@@ -2,173 +2,84 @@
-            largeFile.setLength(4 * 1024 * 1024 * 1024 + 2);
+            RandomAccessFile largeFile = new RandomAccessFile(hugeFile.toFile(), "rw");
+            largeFile.setLength(length);
         } catch (IOException e) {
-            /* We probably don't have enough free disk space
-             * That's ok, we'll skip this test...
-             */
-            LOGGER.log(Level.SEVERE, "Couldn't set up huge file for huge64bitFile test", e);
-            return;
+            // We probably don't have enough free disk space. That's ok, we'll skip this test...
+            Assume.assumeNoException(e);
         }
 
         // a file to store the zip archive in
-        File zipFile = null;
+        Path zipFile = Files.createTempFile(tmp.getRoot().toPath(), "test", ".zip");
 
         // create zip from tmpDir
-        ZipArchiver archiver = null;
-
-        try {
-            zipFile = File.createTempFile("test", ".zip");
-            archiver = new ZipArchiver(Files.newOutputStream(zipFile.toPath()));
-
-            archiver.visit(hugeFile, "huge64bitFileTest.txt");
-        } catch (Exception e) {
-            fail("exception driving ZipArchiver", e);
-        } finally {
-            if (archiver != null) {
-                try {
-                    archiver.close();
-                } catch (IOException e) {
-                    // ignored
-                }
-            }
+        try (ZipArchiver archiver = new ZipArchiver(Files.newOutputStream(zipFile))) {
+            archiver.visit(hugeFile.toFile(), "huge64bitFileTest.txt");
         }
 
         // examine zip contents and assert that there's an item there...
-        String zipEntryName = null;
-
-        try (ZipFile zipFileVerify = new ZipFile(zipFile)) {
-
-            zipEntryName = ((ZipEntry) zipFileVerify.entries().nextElement()).getName();
-        } catch (Exception e) {
-            fail("failure enumerating zip entries", e);
-        }
-
-        assertEquals("huge64bitFileTest.txt", zipEntryName);
-    }
-
-    /**
-     * Convenience method for failing with a cause.
-     *
-     * @param msg the failure description
-     * @param cause the root cause of the failure
-     */
-    private void fail(final String msg, final Throwable cause) {
-        LOGGER.log(Level.SEVERE, msg, cause);
-        Assert.fail(msg);
-    }
-
-    /**
-     * Recursively deletes a directory and all of its children.
-     *
-     * @param f the File (neé, directory) to delete
-     */
-    private void deleteDir(final File f) {
-        for (File c : f.listFiles()) {
-            if (c.isDirectory()) {
-                deleteDir(c);
+        try (ZipFile zipFileVerify = new ZipFile(zipFile.toFile())) {
+            assertEquals(1, zipFileVerify.size());
+            ZipEntry zipEntry = zipFileVerify.entries().nextElement();
+            assertEquals("huge64bitFileTest.txt", zipEntry.getName());
+            assertEquals(length, zipEntry.getSize());
```

Comment:
Unlike before, we are now asserting that the file is of the expected length within the ZIP64 archive.